File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    people.sort((a,b) => a-b)
    // console.log(people)
    let heavy=people.length-1
    let light=0
    while (light<=heavy){
        if (people[heavy] + people[light]>limit ){
            heavy-=1
            answer+=1
        }else{
            heavy-=1
            light+=1
            answer+=1
        }
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py

def solution(people, limit):
    answer = 0

    people = sorted(people)
    print(people)
    l = 0
    r = len(people)-1

    while l <= r:
        if people[l] + people[r] > limit:
            answer += 1
            r -= 1
        else:
            answer += 1
            r -= 1
            l += 1

    return answer
